src/pyqasm/visitor.py

```diff
@@ -2186,6 +2240,10 @@ def visit_statement(self, statement: qasm3_ast.Statement) -> list[qasm3_ast.Stat
             qasm3_ast.SubroutineDefinition: self._visit_subroutine_definition,
             qasm3_ast.ExpressionStatement: lambda x: self._visit_function_call(x.expression),
             qasm3_ast.IODeclaration: lambda x: [],
+            qasm3_ast.WhileLoop: self._visit_while_loop,
+            # Add break/continue if present in AST
+            getattr(qasm3_ast, 'BreakStatement', None): self._visit_break_statement,
```

Comment:
Since `break` and `continue` raise syntax error when existing out of loops, we will never encounter a case where the AST will contain them independently. So, we can remove these checks